packages/trader/src/Stores/Modules/Trading/Helpers/chart.ts

```diff
@@ -0,0 +1,233 @@
+    const {
+        is_favorite,
+        is_open,
+        symbol_category: tab_market_name = '',
+        search_string,
+        symbol: market_type_name = '',
+    } = option;
+    const market_event_type = 'ce_market_types_form';
+    const favorites_action = is_favorite ? ACTION.ADD_TO_FAVORITES : ACTION.DELETE_FROM_FAVORITES;
+    const open_close_action = is_open ? ACTION.OPEN : ACTION.CLOSE;
+    if (option.search_string && state === STATE_TYPES.MARKET_SEARCH) {
+        return {
+            data: {
+                action: ACTION.SEARCH,
+                search_string,
+            },
+            event_type: market_event_type,
+        };
+    }
+    if (state === STATE_TYPES.MARKETS_LIST_TOGGLE) {
+        return {
+            data: {
+                action: open_close_action,
+                market_type_name,
+            },
+            event_type: market_event_type,
+        };
+    }
+    if (state === STATE_TYPES.FAVORITE_MARKETS_TOGGLE && market_type_name) {
+        return {
+            data: {
+                action: favorites_action,
+                market_type_name,
+            },
+            event_type: market_event_type,
+        };
+    }
+    if (state === STATE_TYPES.SYMBOL_CHANGE) {
+        return {
+            data: {
+                action: ACTION.CHOOSE_MARKET_TYPE,
+                market_type_name,
+                tab_market_name,
+            },
+            event_type: market_event_type,
+        };
+    }
+    if (state === STATE_TYPES.MARKET_INFO_REDIRECT) {
+        return {
+            data: {
+                action: ACTION.INFO_REDIRECT,
+                tab_market_name,
+            },
+            event_type: market_event_type,
+        };
+    }
+    return {};
+};
+
+export const getChartAnalyticsData = (state: keyof typeof STATE_TYPES, option: TStateChangeOption) => {
+    const chart_type_form_events: string[] = [
+        STATE_TYPES.CHART_INTERVAL_CHANGE,
+        STATE_TYPES.CHART_MODE_TOGGLE,
+        STATE_TYPES.CHART_TYPE_CHANGE,
+    ];
+    const indicator_type_form_events: string[] = [
+        STATE_TYPES.INDICATOR_ADDED,
+        STATE_TYPES.INDICATOR_DELETED,
+        STATE_TYPES.INDICATOR_INFO_TOGGLE,
+        STATE_TYPES.INDICATOR_SEARCH,
+        STATE_TYPES.INDICATOR_SETTINGS_OPEN,
+        STATE_TYPES.INDICATORS_CLEAR_ALL,
+        STATE_TYPES.INDICATORS_MODAL_TOGGLE,
+    ];
+    const market_type_form_events: string[] = [
+        STATE_TYPES.FAVORITE_MARKETS_TOGGLE,
+        STATE_TYPES.MARKET_INFO_REDIRECT,
+        STATE_TYPES.MARKETS_LIST_TOGGLE,
+        STATE_TYPES.MARKET_SEARCH,
+        STATE_TYPES.SYMBOL_CHANGE,
+    ];
+    if (chart_type_form_events.includes(state)) return getChartTypeFormAnalyticsData(state, option);
+    if (indicator_type_form_events.includes(state)) return getIndicatorTypeFormAnalyticsData(state, option);
+    if (market_type_form_events.includes(state)) return getMarketTypeFormAnalyticsData(state, option);
+    return {};
```

Comment:
I've separated `getChartAnalyticsData` into helpers for each form since we are planning to add more events for other forms soon, so we need more structured approach.